File: src-tauri/src/entities/stat.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Debug, Clone, Deserialize)]
pub struct Stat {
    pub nb_folders: u32,
    pub nb_files: u32,
    pub total_size: u64,
    pub indexed_files: u32,
    pub unindexed_files: u32,
    pub content_indexed_files: u32,
    pub uncontent_indexed_files: u32,
    pub unindexable_files: u32,
    pub indexed_percentage: f64,
    pub content_indexed_percentage: f64,
}

File: src/types/stat.ts
export interface Stat {
    nb_folders: number;
    nb_files: number;
    total_size: number;
    indexed_files: number;
    unindexed_files: number;
    indexed_percentage: number;
    content_indexed_files: number;
    uncontent_indexed_files: number;
    content_indexed_percentage: number;
    unindexable_files: number;
}